src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceCommand.cs

```diff
@@ -0,0 +1,201 @@
+        public SwitchParameter AsJob { get; set; }
+
+        /// <summary>
+        /// Gets or sets the target.
+        /// </summary>
+        /// <value>The target.</value>
+        protected override string Target => Name;
+
+        /// <summary>
+        /// Gets or sets the action message.
+        /// </summary>
+        /// <value>The action message.</value>
+        protected override string ActionMessage => $"{StorageSyncResources.SetStorageSyncServiceActionMessage} {Name}";
+
+        /// <summary>
+        /// Executes the cmdlet.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+
+            base.ExecuteCmdlet();
+
+            ExecuteClientAction(() =>
+            {
+                var resourceName = default(string);
+                var resourceGroupName = default(string);
+
+                // Handle ResourceId Parameter Set
+                if (this.IsParameterBound(c => c.ResourceId))
+                {
+                    var resourceIdentifier = new ResourceIdentifier(ResourceId);
+                    resourceName = resourceIdentifier.ResourceName;
+                    resourceGroupName = resourceIdentifier.ResourceGroupName;
+                }
+                else if (this.IsParameterBound(c => c.InputObject))
+                {
+                    resourceName = InputObject.StorageSyncServiceName;
+                    resourceGroupName = InputObject.ResourceGroupName;
+                }
+                else
+                {
+                    resourceName = Name;
+                    resourceGroupName = ResourceGroupName;
+                }
+
+                string incomingTrafficPolicy;
+                if (this.IsParameterBound(c => c.IncomingTrafficPolicy))
+                {
+                    if(string.IsNullOrEmpty(this.IncomingTrafficPolicy))
+                    {
+                        throw new PSArgumentException(nameof(IncomingTrafficPolicy));
+                    }
+                    incomingTrafficPolicy = this.IncomingTrafficPolicy;
+                }
+                else
+                {
+                    incomingTrafficPolicy = StorageSyncModels.IncomingTrafficPolicy.AllowAllTraffic;
+                }
+
+                var updateParameters = new StorageSyncServiceUpdateParameters()
+                {
+                    Tags = TagsConversionHelper.CreateTagDictionary(Tag ?? new Hashtable(), validate: true),
+                    IncomingTrafficPolicy = incomingTrafficPolicy
+                };
+
+                Target = string.Join("/", resourceGroupName, resourceName);
+                if (ShouldProcess(Target, ActionMessage))
+                {
+                    StorageSyncModels.StorageSyncService storageSyncService = StorageSyncClientWrapper.StorageSyncManagementClient.StorageSyncServices.Update(ResourceGroupName, Name, updateParameters);
+
+                    WriteObject(storageSyncService);
```

Comment:
should this be transformed to PSStorageSyncService

Reply:
its a custom function